src/main/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteTypeMerger.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.hmpps.casenotes.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.justice.hmpps.casenotes.dto.CaseNoteType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class CaseNoteTypeMerger {
+
+    public List<CaseNoteType> mergeAndSortList(final List<CaseNoteType> list1, final List<CaseNoteType> list2) {
+        return merge(list1, list2)
+                .stream()
+                .map(t -> CaseNoteType.builder()
+                            .code(t.getCode())
+                            .description(t.getDescription())
+                            .activeFlag(t.getActiveFlag())
+                            .subCodes(t.getSubCodes().stream().sorted().collect(Collectors.toList()))
+                            .build())
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    private List<CaseNoteType> merge(final List<CaseNoteType> list1, final List<CaseNoteType> list2) {
+
+        final var map1 = list1 != null ? list1.stream().collect(Collectors.toMap(CaseNoteType::getCode, cn -> cn)) : new HashMap<String, CaseNoteType>();
+        final var map2 = list2 != null ? list2.stream().collect(Collectors.toMap(CaseNoteType::getCode, cn -> cn)) : new HashMap<String, CaseNoteType>();
+
+        final var mergedMap = Stream.of(map1, map2)
```

Comment:
do you need to do the collect in to map above? 
could you not just stream each list as you will still have access to the key `getCode` and the value anyway. 
(would still need the null checks though)